### PR TITLE
[OBSDEF-3306] [OBSDEF-3410] Added loading spinner in wizard step and fix console errors

### PR DESCRIPTION
**Pull request by akate1**
**Description :** 
 - Added loading spinner in wizard step
- Fixed below console errors on datagrid and select component

console errors: 
`Warning: Can't call setState on a component that is not yet mounted. This is a no-op, but it might indicate a bug in your application. Instead, assign to `this.state` directly or define a `state = {};` class property with the desired state in the DataGrid component.
    in DataGrid (at DataGrid.stories.tsx:88)
    in div (at DataGrid.stories.tsx:87)
    in div (created by Story)`

`react-dom.development.js:88 Warning: Use the `defaultValue` or `value` props on <select> instead of setting `selected` on <option>.
    in option (at DataGrid.tsx:1147)
    in select (at DataGrid.tsx:1140)`

**JIRA:**
- https://jira.cec.lab.emc.com/browse/OBSDEF-3306
- https://jira.cec.lab.emc.com/browse/OBSDEF-3410

**Testing:** 
image

image

image

### CODE REVIEW

File: src/datagrid/DataGrid.tsx [hunk 1]
```diff
@@ -228,6 +228,18 @@ export enum GridRowType {
 }
 
 const isSelectedKey = "isSelected";
+
+// Default label for datagrid items
+const DEFAULT_ITEM_TEXT: string = "items";
```
Comment (maheshkulkarni4) [inline src/datagrid/DataGrid.tsx:233]: Why do we need defauly text ?  isnt this something which should come as Prop just like other labels ?

File: src/forms/select/Select.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ import {Icon} from "../../icon";
 import {classNames} from "../../utils";
 
 type SelectOption = {
-    value?: string;
+    value?: any;
```
Comment (maheshkulkarni4) [inline src/forms/select/Select.tsx:17]: Why are we switching from String to any ?  Any opens door for more problems as strict checking is close to disabled in this case.

File: src/newWizard/WizardStep.tsx [hunk 3]
```diff
@@ -52,6 +53,7 @@ export default class WizardStep extends React.PureComponent<WizardStepProps> {
                 aria-hidden={id !== currentStepID}
                 className={classNameList}
             >
+                {isLoading && <Spinner size={SpinnerSize.MEDIUM} />}
```
Comment (maheshkulkarni4) [inline src/newWizard/WizardStep.tsx:56]: Just thinking, do we need the spinnerSize to be a Prop instead ?

File: src/datagrid/DataGrid.tsx [hunk 4]
```diff
@@ -312,6 +293,76 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
...
+
+            const firstItem: number = this.getFirstItemIndex(currentPageNumber, datagridPageSize);
+            const lastItem: number = this.getLastItemIndex(datagridPageSize, totalItemsInDatagrid, firstItem);
+
+            const paginationState: DataGridPaginationState = {
+                currentPage: currentPageNumber,
+                pageSize: datagridPageSize,
+                totalItems: totalItemsInDatagrid,
+                pageSizes: pageSizes,
+                compactFooter: compactFooter || false,
```
Comment (maheshkulkarni4) [inline src/datagrid/DataGrid.tsx:356]: Might be old code, but why is this called CompactFooter and why not Footer ?
do we have Footer(big) as well ?

File: src/datagrid/DataGrid.tsx [hunk 5]
```diff
@@ -312,6 +293,76 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
...
+    };
+
+    // Function to initialize rows data in state
+    initializeRowData = (): DataGridRow[] => {
+        const {rows} = this.props;
+        let updatedRows: DataGridRow[] = [];
+        if (rows && rows.length) {
+            updatedRows = this.updateRowIDs(rows);
+            updatedRows.forEach(function(row) {
+                const rowSelectionIsDisabled = row.disableRowSelection !== undefined ? row.disableRowSelection : false;
```
Comment (maheshkulkarni4) [inline src/datagrid/DataGrid.tsx:319]: **row.disableRowSelection !== undefined  ? ** 
can be replaced with 
**row.disableRowSelection ?**  as well.

**Review by maheshkulkarni4 — APPROVED** [review]
LGTM

File: src/datagrid/DataGrid.tsx (hunk 4 above)
Comment (akate1) [reply src/datagrid/DataGrid.tsx:356]: @maheshkulkarni4 This is for compact footer which we show when objectStoreDetails page is open

image

File: src/datagrid/DataGrid.tsx (hunk 5 above)
Comment (akate1) [reply src/datagrid/DataGrid.tsx:319]: @maheshkulkarni4 `disableRowSelection `is a `Boolean `field So here I wanted to check if user has passed it or not.

File: src/forms/select/Select.tsx (hunk 2 above)
Comment (akate1) [reply src/forms/select/Select.tsx:17]: @maheshkulkarni4  `Value` prop has `any` type in Select component and here it was string. So I changed it to any. With this we can create Select of type number , string or any other options. Let me know your thoughts on this.

File: src/datagrid/DataGrid.tsx (hunk 1 above)
Comment (akate1) [reply src/datagrid/DataGrid.tsx:233]: @maheshkulkarni4 This it default text that will get displayed in footer if user has not send any label to display. 
For example, 10 items vs 10 users. Should I make it a mandatory prop ?

File: src/newWizard/WizardStep.tsx (hunk 3 above)
Comment (akate1) [reply src/newWizard/WizardStep.tsx]: done

File: src/datagrid/DataGrid.tsx [hunk 6]
```diff
@@ -312,6 +293,76 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
         }
     }
 
+    // funtion to initialize datagrid state
```
Comment (payald) [inline src/datagrid/DataGrid.tsx:296]: Typo funtion -> function

File: src/datagrid/DataGrid.tsx [hunk 7]
```diff
@@ -312,6 +293,76 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
...
+    initializeDataGridState = (): DataGridState => {
+        const {isLoading, itemText} = this.props;
+        const rows = this.initializeRowData();
+        const columns = this.initializeColumnData();
+        const state: DataGridState = {
```
Comment (payald) [inline src/datagrid/DataGrid.tsx:301]: Can we rename state to dataGridState?

Comment (akate1) [reply src/datagrid/DataGrid.tsx]: done

File: src/datagrid/DataGrid.tsx (hunk 6 above)
Comment (akate1) [reply src/datagrid/DataGrid.tsx]: done

**Review by payald — APPROVED** [review]

**Pull request merged**